provider/vsphere/internal/vsphereclient/createvm.go

```diff
@@ -402,24 +402,48 @@ func (c *Client) selectDatastore(
 	if err := c.client.Retrieve(ctx, refs, nil, &datastores); err != nil {
 		return nil, errors.Annotate(err, "retrieving datastore details")
 	}
+
+	if len(datastores) == 0 {
+		c.logger.Errorf("no datastores available")
+		return nil, errors.New("no datastores available")
+	}
+
 	if args.Constraints.RootDiskSource != nil {
 		dsName := *args.Constraints.RootDiskSource
 		c.logger.Debugf("desired datasource %q", dsName)
 		for _, ds := range datastores {
-			c.logger.Debugf("seen %q", ds.Name)
+			c.logger.Debugf("seen %s", ds.Name)
 			if ds.Name == dsName {
+				c.logger.Infof("selecting datastore %s", ds.Name)
 				return &ds, nil
 			}
 		}
 		return nil, errors.Errorf("could not find datastore %q", dsName)
 	}
+
+	var accessibleDatastores []mo.Datastore
 	for _, ds := range datastores {
 		if ds.Summary.Accessible {
-			c.logger.Debugf("using datastore %q", ds.Name)
-			return &ds, nil
+			accessibleDatastores = append(accessibleDatastores, ds)
+		} else {
+			c.logger.Debugf("datastore %s is inaccessible", ds.Name)
 		}
 	}
-	return nil, errors.New("could not find an accessible datastore")
+
+	if len(accessibleDatastores) == 1 {
+		ds := accessibleDatastores[0]
+		c.logger.Infof("selecting datastore %s", ds.Name)
+		return &ds, nil
+	} else if len(accessibleDatastores) > 1 {
+		var datastoreNames []string
+		for _, ds :=range accessibleDatastores {
+			datastoreNames = append(datastoreNames, ds.Name)
+		}
+		c.logger.Errorf("no datastore provided, select one from %q", strings.Join(datastoreNames, ", "))
```

Comment:
Why not put this in the returned error?

Reply:
Mainly because I see log messages as for the end user and errors for developers, but perhaps that's not entirely correct.

Reply:
Well, the error will be what the user sees if it stops the bootstrap. In fact, we shouldn't be logging this here, otherwise they'll see the message twice.

Reply:
> Mainly because I see log messages as for the end user and errors for developers, but perhaps that's not entirely correct.

It's the exact opposite - errors are for end users because that's what they see. Log messages are for developers because they are the ones that trawl through logs.